man/fi_mr.3.md

```diff
@@ -219,7 +219,24 @@ The following apply to memory registration.
 
 *FI_MR_ALLOCATED*
 : When set, all registered memory regions must be backed by physical
-  memory pages at the time the registration call is made.
+  memory pages at the time the registration call is made. In addition,
+  applications must not perform operations which may result in the underlying
+  virtual address to physical page mapping to change (e.g. calling free()
+  against an allocated MR). Failing to adhere to this may result in the
+  virtual address pointing to one set of physical pages while the MR points
+  to another set of physical pages.
+
+  Note: The operating system deciding to migrate application physical pages
+  may result in the virtual address pointing to one set of physical pages while
+  the MR points to another set of physical pages. Operating system features
+  which may do this should be disabled.
```

Comment:
This note needs to be reworded or dropped.  The entire point of virtual addressing is to allow the physical pages behind them to change.  You can't 'disable' this feature without turning off the OS.

Reply:
Without ODP, RDMA requires this does not change otherwise you risk potential data corruption. In Linux, with HPC at least, there are knobs to disable common things (THP and NUMA balancing) to prevent memory migration.

If we take"disable" this feature meaning turn off the OS, at face value, this means the MR cache + memhooks should never be used since memhooks would not be able to capture this.

Would it be better to state which Linux features may cause this and suggest disabling?

Reply:
You need to be clear what you mean by migrate.  Virtual memory exists to allow the physical pages to change.

Reply:
RDMA without ODP requires physical pages to not change while registered.

I ended up just deleting this documentation.